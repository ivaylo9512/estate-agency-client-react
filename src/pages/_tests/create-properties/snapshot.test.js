import { shallow } from 'enzyme';
import CreateProperty from 'pages/create-property';
import * as Redux from 'react-redux';

describe('create properties snapshot tests', () => {
    let selectorSpy;

    beforeAll(() => {
        jest.spyOn(Redux, 'useDispatch').mockReturnValue(jest.fn());

        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (errors) => {
        selectorSpy.mockReturnValue(errors);

        return shallow(
            <CreateProperty />
        )
    }
    
    it('should match snapshot with errors', () => {
        const wrapper = createWrapper({ isLoading: false, error: { name: 'name error', price: 'price error', location: 'location error', size: 'size error', description: 'description error' }});

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot without errors', () => {
        const wrapper = createWrapper({ isLoading: false });

        expect(wrapper).toMatchSnapshot();
    })
})