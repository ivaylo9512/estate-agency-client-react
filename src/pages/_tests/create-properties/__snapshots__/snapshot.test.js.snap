// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create properties snapshot tests should match snapshot with errors 1`] = `
<section>
  <h1>
    CreateProperty
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name:
    </label>
    <InputWithError
      error="name error"
      input={
        <input
          name="name"
          onChange={[Function]}
          placeholder="name"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="price"
    >
      Price:
    </label>
    <InputWithError
      error="price error"
      input={
        <input
          name="price"
          onChange={[Function]}
          placeholder="price"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="location"
    >
      Location:
    </label>
    <InputWithError
      error="location error"
      input={
        <input
          name="location"
          onChange={[Function]}
          placeholder="location"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="size"
    >
      Size:
    </label>
    <InputWithError
      error="size error"
      input={
        <input
          name="size"
          onChange={[Function]}
          placeholder="size"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="description"
    >
      description:
    </label>
    <InputWithError
      error="description error"
      input={
        <input
          name="description"
          onChange={[Function]}
          placeholder="description"
          required={true}
          type=""
          value=""
        />
      }
    />
    <select
      onChange={[Function]}
      value={1}
    >
      <option>
        1
      </option>
      <option>
        2
      </option>
      <option>
        3
      </option>
      <option>
        4
      </option>
      <option>
        5
      </option>
      <option>
        6
      </option>
      <option>
        7
      </option>
    </select>
    <button>
      submit
    </button>
  </form>
</section>
`;

exports[`create properties snapshot tests should match snapshot without errors 1`] = `
<section>
  <h1>
    CreateProperty
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Name:
    </label>
    <InputWithError
      input={
        <input
          name="name"
          onChange={[Function]}
          placeholder="name"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="price"
    >
      Price:
    </label>
    <InputWithError
      input={
        <input
          name="price"
          onChange={[Function]}
          placeholder="price"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="location"
    >
      Location:
    </label>
    <InputWithError
      input={
        <input
          name="location"
          onChange={[Function]}
          placeholder="location"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="size"
    >
      Size:
    </label>
    <InputWithError
      input={
        <input
          name="size"
          onChange={[Function]}
          placeholder="size"
          required={true}
          type=""
          value=""
        />
      }
    />
    <label
      htmlFor="description"
    >
      description:
    </label>
    <InputWithError
      input={
        <input
          name="description"
          onChange={[Function]}
          placeholder="description"
          required={true}
          type=""
          value=""
        />
      }
    />
    <select
      onChange={[Function]}
      value={1}
    >
      <option>
        1
      </option>
      <option>
        2
      </option>
      <option>
        3
      </option>
      <option>
        4
      </option>
      <option>
        5
      </option>
      <option>
        6
      </option>
      <option>
        7
      </option>
    </select>
    <button>
      submit
    </button>
  </form>
</section>
`;
